packages/pwa/webpack.config.js

```diff
@@ -66,16 +70,78 @@ module.exports = {
             PL_TERMS_OF_SERVICE: terms_of_service,
         }),
         new CleanWebpackPlugin(),
+        {
+            apply(compiler) {
+                compiler.hooks.compilation.tap("Store Built Files for CSP", (compilation) => {
+                    // We tap into this hook to make sure we have the array populated before the CspHtmlWebpackPlugin runs
+                    HtmlWebpackPlugin.getHooks(compilation).beforeEmit.tapAsync(
+                        "Store Built Files for CSP",
+                        (data, callback) => {
+                            compilation.chunks.forEach((chunk) => {
+                                builtFilesForCsp.push(...chunk.files);
+                            });
+
+                            callback(null, data);
+                        }
+                    );
+
+                    return true;
+                });
+            },
+        },
         new HtmlWebpackPlugin({
             title: name,
             template: resolve(__dirname, "src/index.html"),
             meta: {
                 "Content-Security-Policy": {
                     "http-equiv": "Content-Security-Policy",
-                    content: `default-src 'self' ${serverUrl} https://api.pwnedpasswords.com blob:; style-src 'self' 'unsafe-inline'; object-src 'self' blob:; frame-src 'self'; img-src 'self' blob: data: https:;`,
+                    content: `default-src 'self' ${serverUrl};`, // NOTE: This will be overwritten below, but we need the tag to exist
```

Comment:
Can you explain what we need `@melloware/csp-webpack-plugin` for? Or in other words, why don't we just build that string ourselves?

Reply:
Doing the hashing + nonces + type safety for the CSP is not super easy or trivial work. This is also _only_ a dev dependency, so it's not going to impact customers or the end product.

Reply:
Got it. Let's just make sure to properly audit the dependency before using it in production.

Reply:
It's not really used in "production", only while building the app. If you think that's a problem, I'm happy to just not do the nonces and hashes (which does ensure a lot more security), which will remove the need for the dependency.

Reply:
I just ran into a problem with the hashes nonces. Since they'll vary on every build, we can't then reliably use checksums as they'll fail for `index.html` and `main.js` when trying to compare locally. I'll remove this dependency and manually build the CSP.